flink-staging/flink-scala-shell/src/test/scala/org/apache/flink/api/scala/ScalaShellITSuite.scala

```diff
@@ -225,8 +223,50 @@ class ScalaShellITSuite extends FunSuite with Matchers with BeforeAndAfterAll {
     out.toString + stdout
   }
 
-  var cluster: Option[ForkableFlinkMiniCluster] = None
-  val parallelism = 4
+  /**
+   * tests flink shell startup with remote cluster (starts cluster internally)
+   */
+  test("start flink scala shell with remote cluster") {
+
+    val input: String = "val els = env.fromElements(\"a\",\"b\");\n" +
+      "els.print\nError\n:q\n"
+
+    val in: BufferedReader = new BufferedReader(
+      new StringReader(
+        input + "\n"))
+    val out: StringWriter = new StringWriter
+
+    val baos: ByteArrayOutputStream = new ByteArrayOutputStream
+    val oldOut: PrintStream = System.out
+    System.setOut(new PrintStream(baos))
+
+    val (c, args) = cluster match{
+      case Some(cl) =>
+        val arg = Array("remote",
+          cl.hostname,
+          Integer.toString(cl.getLeaderRPCPort))
+        (cl, arg)
+      case None =>
+        fail("Cluster creation failed!")
+    }
+
+    //start scala shell with initialized
+    // buffered reader for testing
+    FlinkShell.bufferedReader = Some(in)
+    FlinkShell.main(args)
+    baos.flush
```

Comment:
Yeah, right. And almost all the method of which return type is `Unit` have some side-effects. :)

> On Oct 8, 2015, at 5:05 PM, Nikolaas Steenbergen <notifications@github.com> wrote:
> 
> In flink-staging/flink-scala-shell/src/test/scala/org/apache/flink/api/scala/ScalaShellITSuite.scala:
> 
> > +
> > +    val (c, args) = cluster match{
> > +      case Some(cl) =>
> > +        val arg = Array("remote",
> > +          cl.hostname,
> > +          Integer.toString(cl.getLeaderRPCPort))
> > +        (cl, arg)
> > +      case None =>
> > +        fail("Cluster creation failed!")
> > +    }
> > +
> > +    //start scala shell with initialized
> > +    // buffered reader for testing
> > +    FlinkShell.bufferedReader = Some(in)
> > +    FlinkShell.main(args)
> > +    baos.flush
> 
> Actually the official reasoning is if it has some "side-effects" it should have some parenthesis (which it does in this case)
> http://docs.scala-lang.org/style/method-invocation.html
> 
> —
> Reply to this email directly or view it on GitHub.
> 

Regards,
Chiwan Park